impl/src/expand.rs

```diff
@@ -377,6 +547,132 @@ fn impl_enum(input: Enum) -> TokenStream {
     }
 }
 
+#[cfg_attr(test, derive(Debug))]
+#[derive(Clone)]
+struct GenericUsageVisitor {
+    generics: std::collections::HashMap<proc_macro2::Ident, bool>,
+}
+
+impl GenericUsageVisitor {
+    pub fn new<TPairs>(generics: TPairs) -> Self
+    where
+        TPairs: IntoIterator<Item = (syn::Ident, bool)>,
+    {
+        Self {
+            generics: generics.into_iter().collect(),
+        }
+    }
+
+    pub fn new_unmarked<TIdents>(generics: TIdents) -> Self
+    where
+        TIdents: IntoIterator<Item = syn::Ident>,
+    {
+        Self::new(generics.into_iter().map(|ident| (ident, false)))
+    }
+
+    pub fn iter_marked(&self) -> impl Iterator<Item = &proc_macro2::Ident> {
+        self.generics.iter().filter(|(_k, v)| **v).map(|(k, _v)| k)
+    }
+
+    #[allow(dead_code)]
```

Comment:
No dead code please.

Reply:
My bad- that one got left behind- I must've forgotten to tag it `todo` and wasn't clear I'd use it or not before the feature was ready.